bycycle/burst/utils.py

```diff
@@ -49,3 +49,73 @@ def check_min_burst_cycles(is_burst, min_n_cycles=3):
             temp_cycle_count = 0
 
     return is_burst
+
+
+def recompute_edges(df_features, threshold_kwargs):
+    """Recompute the is_burst column for cycles on the edges of bursts.
+
+    Parameters
+    ----------
+    df_features : pandas.DataFrame
+        A dataframe containing shape and burst features for each cycle.
+    threshold_kwargs : dict, optional, default: None
+        Feature thresholds for cycles to be considered bursts, matching keyword arguments for:
+
+        - :func:`~.detect_bursts_cycles` for consistency burst detection
+          (i.e. when burst_method == 'cycles')
+
+    Returns
+    -------
+    df_features_edges : pandas.DataFrame
+        An cycle feature dataframe with an updated ``is_burst`` column for edge cycles.
+
+    Notes
+    -----
+
+    - `df_features` must be computed using consistency burst detection.
+
+    Examples
+    --------
+    Lower the amplitude consistency threshold to zero for cycles on the edges of bursts:
+
+    >>> sig = sim_combined(n_seconds=4, fs=1000, components={'sim_bursty_oscillation': {'freq': 10},
+    ...                                                      'sim_powerlaw': {'exp': 2}})
+    >>> threshold_kwargs = {'amp_fraction_threshold': 0., 'amp_consistency_threshold': .5,
+    ...                     'period_consistency_threshold': .5, 'monotonicity_threshold': .4,
+    ...                     'min_n_cycles': 3}
+    >>> df_features, _ = compute_features(sig, fs=1000, f_range=(8, 12), threshold_kwargs=threshold_kwargs)
+    >>> threshold_kwargs['amp_consistency_threshold'] = 0
+    >>> df_features_edges = recompute_edges(df_features, threshold_kwargs)
+    """
+
+    from bycycle.burst import detect_bursts_cycles
+
+    # Prevent overwriting the orignal dataframe
+    df_features_edges = df_features.copy()
+
+    # Identify all cycles where is_burst changes on the following cycle
+    #   Use copy to keep dataframe columns unlinked
+    is_burst = df_features_edges.copy()['is_burst'].values
```

Comment:
This is relatively minor - but this line computes a copy on the whole dataframe, can it not instead compute a copy of the extracted column, either as something like:
`df_features_edges['is_burst'].copy().values`
or
`from copy import deepcopy`
`deepcopy(df_features_edges.['is_burst'].values`)

These df's could presumably be quite large, and we already compute a copy of the whole df above (line 94). Copying the whole df twice (storing 3 copies) seems like probably more than we need.

Reply:
Agreed and thanks for the suggestion! A single deepcopy of just the values (as in your example) was all that was need, rather than 3 copy's.